ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/ServiceToOpenAPIMapper.java

```diff
@@ -222,6 +233,24 @@ public static OASResult generateOAS(OASGenerationMetaInfo oasGenerationMetaInfo)
         }
     }
 
+    private static boolean isInterceptableService(ServiceDeclarationNode serviceDefinition,
+                                                  SemanticModel semanticModel) {
+        AtomicBoolean isInterceptable = new AtomicBoolean(false);
```

Comment:
why do we need an AtomicBoolean here? Do we have concurrent accesses for this?

Reply:
This is since I am using a lambda expression to check whether the service is a interceptable service. If I can change this to `boolean[]` then it will work. Will change like that 👍🏼

Reply:
Fixed via [b7b157e](https://github.com/ballerina-platform/openapi-tools/pull/1614/commits/b7b157e80d2574a5f1708c915c92dad05bb532d2)